io/EptReader.cpp

```diff
@@ -57,11 +60,64 @@ namespace
     };
 
     const std::string addonFilename { "ept-addon.json" };
+
+    Dimension::Type getType(const NL::json& dim)
+    {
+        if (dim.contains("scale"))
+            return Dimension::Type::Double;
+
+        const std::string typestring(dim["type"].get<std::string>());
```

Comment:
Using `.at()` would be safer here than `[]`.

Reply:
How so?  `.at()` throws.

Reply:
If the key doesn't exist, `[]` potentially segfaults or reads garbage.  An exception seems safer than UB.

Reply:
I suppose we always *should* have a type, so it's an error if it isn't present, right?

Reply:
Correct.  But for bad data where it's trivial to create an exception rather than a segfault, it seems preferable to do so.

Reply:
Agreed.  Just wondering if I should do something better.